From 67b3610542f5b9ca4c87d46b2a02173025837c43 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Tue, 15 Nov 2016 16:32:29 -0500
Subject: [PATCH 107/134] trace-cmd: Fix file pointer leak on error exit of
 tracecmd_find_tracing_dir()

If the strdup() fails to allocate, we exit tracecmd_find_tracing_dir()
without closing the file descriptor just opened.

Reported-by: David Binderman <dcb314@hotmail.com>
Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-util.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/trace-util.c b/trace-util.c
index 5bf591c0f458..14656d9aaaa7 100644
--- a/trace-util.c
+++ b/trace-util.c
@@ -723,8 +723,10 @@ char *tracecmd_find_tracing_dir(void)
 			break;
 		if (!debug_str && strcmp(type, "debugfs") == 0) {
 			debug_str = strdup(fspath);
-			if (!debug_str)
+			if (!debug_str) {
+				fclose(fp);
 				return NULL;
+			}
 		}
 	}
 	fclose(fp);
-- 
2.5.5

